service/matching/handler/engine.go

```diff
@@ -540,7 +596,22 @@ pollLoop:
 
 			continue pollLoop
 		}
+
 		task.Finish(nil)
+		event.Log(event.E{
+			TaskListName: taskListName,
+			TaskListKind: taskListKind,
+			TaskListType: persistence.TaskListTypeDecision,
+			TaskInfo:     task.Info(),
+			EventName:    "PollForDecisionTask returning task",
+			Host:         e.config.HostName,
+			Payload: map[string]any{
+				"TaskIsForwarded":      task.IsForwarded(),
```

Comment:
we can include the source of task (whether from db or history)

Reply:
good idea. I'll leave that to a follow up PR